presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -288,14 +289,28 @@ public void copyLogsToHostPath(Path hostPath)
         }
     }
 
-    public DockerContainer waitingForAll(WaitStrategy... strategies)
+    @Override
+    public DockerContainer waitingFor(WaitStrategy waitStrategy)
```

Comment:
I think the semantics is cumbersome. 
`waitFor` sometimes overrides sometimes adds. 
`watiForAll` always overrides.

I would prefere that override strategy would be explicit in the call. Either in method name, or as an explicit extra parameter.